### PR TITLE
Thomas survey detail test

### PR DESCRIPTION
**Pull request by TKostrzewski**
Added a test for the entire survey detail page

### CODE REVIEW

File: client/src/pages/SurveyDetail/SurveyDetail.test.jsx [hunk 1]
```diff
@@ -1,29 +1,164 @@
...
+    const mockAxiosGetFirst = jest.spyOn(axios, 'get');
+    mockAxiosGetFirst.mockImplementation((url) => {
+      if (url === '/employees') {
+        return Promise.resolve({ data: dummySurvey.recipients });
+      }
+      return Promise.resolve({ data: dummySurvey });
+    });
+
+    const mockAxiosPatch = jest.spyOn(axios, 'patch');
+    mockAxiosPatch.mockImplementation(() => Promise.resolve({}));
```
Comment (AntonioGargaro) [inline client/src/pages/SurveyDetail/SurveyDetail.test.jsx]: Good work with mocks!

File: client/src/pages/SurveyDetail/SurveyDetail.test.jsx [hunk 2]
```diff
@@ -1,29 +1,164 @@
...
+    ).toBeVisible();
+
+    expect(getByTestId('survey-detail-stepper')).toBeVisible();
+
+    expect(
+      getByText(
+        `${dummySurveyPublished.responses.length}/${dummySurveyPublished.recipients.length}`,
+      ),
+    ).toBeVisible();
+    expect(getByText('respondents')).toBeVisible();
```
Comment (AntonioGargaro) [inline client/src/pages/SurveyDetail/SurveyDetail.test.jsx]: You use a lot of `toBeVisible()` in your tests, but I think `toBeInTheDocument` is more informative. Not important to change though.

File: client/src/pages/SurveyDetail/SurveyDetail.test.jsx [hunk 3]
```diff
@@ -1,29 +1,164 @@
...
 
 describe('Testing the Survey Detail page', () => {
-  it('', (done) => {
-    done();
-    // const { getAllByRole, getByText } = render(
-    //   <Provider store={store}>
-    //     <SurveyDetail />
-    //   </Provider>,
-    // );
+  it('', async () => {
```
Comment (thomas-galligan) [inline client/src/pages/SurveyDetail/SurveyDetail.test.jsx]: can we add a description of the test here?

File: client/src/pages/SurveyDetail/SurveyDetail.test.jsx [hunk 4]
```diff
@@ -1,29 +1,164 @@
...
-    // fireEvent.click(addQuestionButton);
+    expect(
+      getByText(
+        `${dummySurveyClosed.responses.length}/${dummySurveyClosed.recipients.length}`,
+      ),
+    ).toBeVisible();
+    expect(getByText('respondents')).toBeVisible();
 
-    // getByText('Title must be less than 60 characters!', { exact: false });
+    mockAxiosGetThird.mockRestore();
```
Comment (thomas-galligan) [inline client/src/pages/SurveyDetail/SurveyDetail.test.jsx]: this is a fantastically detailed and thorough test!

File: client/src/pages/SurveyDetail/SurveyDetail.test.jsx (hunk 2 above)
Comment (marthaklambert) [reply client/src/pages/SurveyDetail/SurveyDetail.test.jsx]: Using to be in the document after getby text feels kind of like saying expect(true).toBeTrue()

**Pull request merged**